writeaheadlog.go

```diff
@@ -509,45 +462,14 @@ func (w *WAL) wipeWAL() error {
 
 // writeWALMetadata writes WAL metadata to the input file.
 func writeWALMetadata(f file) error {
-	buffer := new(bytes.Buffer)
-
-	// Prepare the data that is written to the buffer
-	recoveryState := uint16(recoveryStateUnclean)
-	headerData := []byte(metadataHeader)
-	headerLength := uint16(len(headerData))
-	versionData := []byte(metadataVersion)
-	versionlength := uint16(len(versionData))
-
-	// Write wipe state
-	if err := binary.Write(buffer, binary.LittleEndian, &recoveryState); err != nil {
-		return build.ExtendErr("could not marshal recovery state", err)
-	}
-	// Write header length
-	if err := binary.Write(buffer, binary.LittleEndian, &headerLength); err != nil {
-		return build.ExtendErr("could not marshal header length", err)
-	}
-	// Write header
-	if _, err := buffer.Write(headerData); err != nil {
-		return build.ExtendErr("could not write header", err)
-	}
-	// Write version length
-	if err := binary.Write(buffer, binary.LittleEndian, &versionlength); err != nil {
-		return build.ExtendErr("could not marshal version length", err)
-	}
-	// Write version
-	if _, err := buffer.Write(versionData); err != nil {
-		return build.ExtendErr("could not write version data", err)
-	}
-	// Sanity check. Metadata must not be greater than pageSize
-	if buffer.Len() > pageSize {
-		panic("WAL metadata is greater than pageSize")
-	}
-	// Write marshalled data to disk
-	_, err := f.WriteAt(buffer.Bytes(), 0)
-	if err != nil {
-		return build.ExtendErr("unable to write WAL metadata", err)
-	}
-	return nil
+	// Create the metadata.
+	data := make([]byte, 0, metadataHeaderSize+metadataVersionSize+metadataStatusSize)
+	data = append(data, []byte(metadataHeader)...)
+	data = append(data, []byte(metadataVersion)...)
+	data = append(data, byte(recoveryStateUnclean))
```

Comment:
you are casting a 2 byte string to 1 byte?

Reply:
no the first byte is the value, and the second byte is a newline.